test/unit/parallel_pusher_test.rb

```diff
@@ -0,0 +1,43 @@
+require 'test_helper'
+
+class ParallelPusherTest < ActiveSupport::TestCase
+  self.use_transactional_tests = false
```

Comment:
This needs to be here because general test flow doesn't clear records created in `Thread.new` which use different db connection.